core/src/dht/dht_store.rs

```diff
@@ -62,16 +63,20 @@ impl DhtStore {
         link_type: Option<String>,
         tag: Option<String>,
     ) -> Result<BTreeSet<EntityAttributeValueIndex>, HolochainError> {
+        // interpret link tags and types as regex
+        let link_type = link_type.map(|s| Regex::new(&s).unwrap());
```

Comment:
Should improve handling of invalid regex and ensure the error happens in the calling code